Runtime error when etcdbr is run with Druid without backups enabled

**How to categorize this issue?**

/kind bug

**What happened**:

Similar to the [issue#53 in etcd-wrapper](https://github.com/gardener/etcd-wrapper/issues/53), I've observed the `backup-restore` container/process going into runtime error when the backups are disabled, I've only observed it without backups. With backups, even if it's a fresh etcd cluster (i.e no existing data directory) , I've not faced any issues as such.

<summary>See the logs</summary>
<details>
  
```
time="2025-05-15T10:40:55Z" level=info msg="etcd-backup-restore Version: v0.36.2"
time="2025-05-15T10:40:55Z" level=info msg="Git SHA: 3c66fe40"
time="2025-05-15T10:40:55Z" level=info msg="Go Version: go1.24.2"
time="2025-05-15T10:40:55Z" level=info msg="Go OS/Arch: linux/arm64"
time="2025-05-15T10:40:55Z" level=info msg="compressionConfig:\n  enabled: false\n  policy: gzip\ndefragmentationSchedule: 0 */24 * * *\netcdConnectionConfig:\n  connectionTimeout: 5m0s\n  defragTimeout: 15m0s\n  endpoints:\n  - http://etcd-test-local:2379\n  insecureSkipVerify: true\n  insecureTransport: true\n  serviceEndpoints:\n  - http://etcd-test-client:2379\n  snapshotTimeout: 15m0s\nexponentialBackoffConfig:\n  attemptLimit: 6\n  multiplier: 2\n  thresholdTime: 2m8s\nhealthConfig:\n  deltaSnapshotLeaseName: delta-snapshot-revisions\n  fullSnapshotLeaseName: full-snapshot-revisions\n  heartbeatDuration: 10s\n  memberGCDuration: 1m0s\n  memberLeaseRenewalEnabled: true\nleaderElectionConfig:\n  etcdConnectionTimeout: 5s\n  reelectionPeriod: 5s\nrestorationConfig:\n  MaxRequestBytes: 10485760\n  MaxTxnOps: 10240\n  autoCompactionMode: periodic\n  autoCompactionRetention: 30m\n  dataDir: /var/etcd/data/new.etcd\n  embeddedEtcdQuotaBytes: 8589934592\n  initialAdvertisePeerURLs:\n  - http://localhost:2380\n  initialCluster: default=http://localhost:2380\n  initialClusterToken: etcd-cluster\n  maxCallSendMsgSize: 10485760\n  maxFetchers: 6\n  name: default\n  tempDir: /var/etcd/data/restoration.temp\nserverConfig:\n  port: 8080\nsnapshotterConfig:\n  deltaSnapshotMemoryLimit: 10485760\n  deltaSnapshotPeriod: 20s\n  deltaSnapshotRetentionPeriod: 0s\n  garbageCollectionPeriod: 1m0s\n  garbageCollectionPolicy: Exponential\n  maxBackups: 7\n  schedule: 0 */1 * * *\nsnapstoreConfig:\n  container: \"\"\n  maxParallelChunkUploads: 5\n  minChunkSize: 5242880\n  prefix: v2\n  tempDir: /var/etcd/data/temp\nuseEtcdWrapper: true\n"
time="2025-05-15T10:40:55Z" level=warning msg="No snapstore storage provider configured. Will not start backup schedule." actor=backup-restore-server
time="2025-05-15T10:40:55Z" level=info msg="Setting status to : 503" actor=backup-restore-server
time="2025-05-15T10:40:55Z" level=info msg="Registering the http request handlers..." actor=backup-restore-server
time="2025-05-15T10:40:55Z" level=info msg="Starting the http server..." actor=backup-restore-server
time="2025-05-15T10:40:55Z" level=info msg="Checking if etcd is running"
time="2025-05-15T10:40:55Z" level=info msg="Starting HTTP server at addr: :8080" actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Responding to status request with: New" actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Received start initialization request." actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Updating status from New to Progress" actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Setting status to : 503" actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Validation failBelowRevision: " actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=info msg="Validation mode: full" actor=backup-restore-server
time="2025-05-15T10:40:56Z" level=warning msg="No snapstore storage provider configured."
time="2025-05-15T10:40:56Z" level=info msg="Removing directory(/var/etcd/data/new.etcd) since snapstore is empty."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1dcc15c]

goroutine 196 [running]:
github.com/gardener/etcd-backup-restore/pkg/initializer.(*EtcdInitializer).Initialize(0x4000872ed0, {0x2625b57, 0x4}, 0x0)
    /go/src/github.com/gardener/backup-restore/pkg/initializer/initializer.go:131 +0x9bc
github.com/gardener/etcd-backup-restore/pkg/server.(*HTTPHandler).serveInitialize.func1()
    /go/src/github.com/gardener/backup-restore/pkg/server/httpAPI.go:258 +0x3c0
created by github.com/gardener/etcd-backup-restore/pkg/server.(*HTTPHandler).serveInitialize in goroutine 191
    /go/src/github.com/gardener/backup-restore/pkg/server/httpAPI.go:230 +0x200
stream closed EOF for default/etcd-test-0 (backup-restore)
```
</details>

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- etcd-backup-restore version: `v0.36.2`
- etcd-druid version: [1fb4ee](https://github.com/gardener/etcd-druid/tree/1fb4ee9e435a0e76c4d0cba8831e34baa08b61e2)

--- a/pkg/initializer/initializer.go
+++ b/pkg/initializer/initializer.go
@@ -128,13 +128,20 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 	// clean up snapshot temp directory and recreate it, since this can be considered part of initializing
 	// the data directory for future snapshotting, if snapshot TempDir is specified.
 	// This also allows cleaning up a previously created temp directory files with wider file permissions.
+
+	if e.Config.SnapstoreConfig == nil {
+		logger.Infof("Will not clean up temporary snapshot directory since no snapstore configured. Continuing.")
+		return nil
+	}
+
 	if e.Config.SnapstoreConfig.TempDir != "" && e.Config.SnapstoreConfig.TempDir != "/tmp" {
 		if err = e.removeDir(e.Config.SnapstoreConfig.TempDir); err != nil {
 			return fmt.Errorf("failed to remove directory %s with err: %v", e.Config.SnapstoreConfig.TempDir, err)
 		}
-		if err = os.MkdirAll(e.Config.SnapstoreConfig.TempDir, 0700); err != nil {
-			return fmt.Errorf("failed to create temporary directory %s: %v", e.Config.SnapstoreConfig.TempDir, err)
-		}
+	}
+	logger.Infof("Creating temporary directory %s if it does not exist.", e.Config.SnapstoreConfig.TempDir)
+	if err = os.MkdirAll(e.Config.SnapstoreConfig.TempDir, 0700); err != nil {
+		return fmt.Errorf("failed to create temporary directory %s: %w", e.Config.SnapstoreConfig.TempDir, err)
 	}
 
 	return nil

